src/sas/sascalc/dataloader/readers/cansas_reader.py

```diff
@@ -1169,11 +1163,13 @@ def _write_notes(self, datainfo, entry_node):
         :param entry_node: lxml node ElementTree object to be appended to
 
         """
-        if len(datainfo.notes) == 0:
-            node = self.create_element("SASnote")
-            self.append(node, entry_node)
-        else:
-            for item in datainfo.notes:
+        from sas.sasview import __version__ as version
+        node = self.create_element("SASnote")
+        self.write_text(
+            node, 'Data file generated by SasView v{0}'.format(version))
+        self.append(node, entry_node)
+        for item in datainfo.notes:
+            if item:
```

Comment:
Do you need the `if item` here?  That is, can  `datainfo.notes` include an empty note?

Reply:
Oddly enough, yes, they can be.